File: src/main/java/be/kdg/kandoe/backend/model/organizations/Topic.java
package be.kdg.kandoe.backend.model.organizations;

import be.kdg.kandoe.backend.model.cards.CardDetails;
import lombok.*;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * A Topic belongs to one category
 * Consists of a name and a description
 * Has a list of cards
 */
@Entity
@Data
@EqualsAndHashCode(exclude = {"category", "cards"})
//@NoArgsConstructor
public class Topic {
    @Id
    @GeneratedValue
    @Setter(AccessLevel.NONE)
    private int topicId;

    @Column(nullable = false)
    private String name;

    private String description;

    @ManyToOne(fetch = FetchType.EAGER)
    private Category category;

    @ManyToMany(fetch = FetchType.EAGER)
    private Set<CardDetails> cards;

    public Topic() {
        this.cards = new HashSet<>();
    }
}

File: src/main/webapp/app/entities/topic.ts
import {Serializable} from "../util/serializable";

export class Topic implements Serializable<Topic>{
    public name : string;
    public description: string;

    constructor(name:string, description:string) {
        this.name = name;
        this.description = description;
    }

    deserialize(object:Topic):Topic {
        this.name = object.name;
        this.description = object.description;

        return this;
    }
}